src/main/kotlin/org/jetbrains/research/ml/codetracker/Plugin.kt

```diff
@@ -1,21 +1,107 @@
 package org.jetbrains.research.ml.codetracker
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
 import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.application.ex.ApplicationEx
 import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import net.lingala.zip4j.ZipFile
+import java.io.File
+import java.io.InputStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+
 
 enum class TestMode {
     ON,
     OFF
 }
 
+/**
+ * Represents a plugin required by Codetracker
+ */
+data class RequiredPlugin(val name: String, val zipFile: String) {
+    private val resourceFolder = "requiredplugins"
+    private val logger: Logger = Logger.getInstance(javaClass)
+
+    /**
+     * Perhaps there is a better way of doing this
+     */
+    fun isInstalled() : Boolean {
+        val isAlreadyInstalled =  File("${PathManager.getPluginsPath()}/${name}").exists()
```

Comment:
I am not sure that's enough. I have macOS system and for each plugin, I have two folders - with the plugin name and with the prefix `plugin` (see an attachment). Should we check all folders all not? Or maybe it is a feature only for macOS users...
![image](https://user-images.githubusercontent.com/8993915/89759496-fe344280-daf2-11ea-9542-518373af5f85.png)

Reply:
It's not about macOS, it's about Activity Tracker. This plugin has an additional folder "activity-tracker" where it stores ide-events.csv and so on. If you look at Activity Tracker gradle file, you'll see an explanation of why it happens.
![image](https://user-images.githubusercontent.com/24303332/89761445-0d1cf400-daf7-11ea-93db-916455d760e7.png)

Reply:
Thank you! I have not known it